eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringMongoReader.java

```diff
@@ -122,4 +179,24 @@ public void close() throws ItemStreamException {
         }
         evaCursor.close();
     }
+
+
+    /**
+     * Retry policy for handling MongoCursorNotFoundException during reads.
+     */
+    class MongoReaderRetryPolicy extends SimpleRetryPolicy {
+
+        public MongoReaderRetryPolicy() {
+            super(MAX_RETRIES, Collections.singletonMap(MongoCursorNotFoundException.class, true));
+        }
+
+        @Override
+        public void registerThrowable(RetryContext context, Throwable throwable) {
+            if (canRetry(context) && context.getRetryCount() < getMaxAttempts()-1
```

Comment:
Since we are not comparing against an array index, I am not sure why we have a n-1 here...

Reply:
The n-1 is so that we don't initialize unnecessarily after the last attempt fails, see e.g. [this test](https://github.com/EBIvariation/eva-accession/pull/316/files/825ca3c8c9843b16d9e272b7bb5ffa7d11209103#diff-658a169121fd1f14103fbfd338d50b93f547b93c82d7ec24c2a81fa06fd044d4R92).  I think it's because the `retryCount` is getting incremented in the superclass's `registerThrowable`.  I'll add a comment explaining this...